allensdk/brain_observatory/behavior/behavior_metadata.py

```diff
@@ -139,3 +147,220 @@ def get_task_parameters(data: Dict) -> Dict:
+
+    @property
+    def session_type(self) -> str:
+        return self._extractor.get_stimulus_name()
+
+    @property
+    def date_of_acquisition(self) -> datetime:
+        """Return the timestamp for when experiment was started in UTC
+
+        NOTE: This method will only get acquisition datetime from
+        extractor (data from LIMS) methods. As a sanity check,
+        it will also read the acquisition datetime from the behavior stimulus
+        (*.pkl) file and raise a warning if the date differs too much from the
+        datetime obtained from the behavior stimulus (*.pkl) file.
+
+        :rtype: datetime
+        """
+        extractor_acq_date = self._extractor.get_date_of_acquisition()
+
+        pkl_data = self._behavior_stimulus_file
+        pkl_raw_acq_date = pkl_data["start_time"]
+        if isinstance(pkl_raw_acq_date, datetime):
+            pkl_acq_date = pytz.utc.localize(pkl_raw_acq_date)
+
+        elif isinstance(pkl_raw_acq_date, (int, float)):
+            # We are dealing with an older pkl file where the acq time is
+            # stored as a Unix style timestamp string
+            parsed_pkl_acq_date = datetime.fromtimestamp(pkl_raw_acq_date)
+            pkl_acq_date = pytz.utc.localize(parsed_pkl_acq_date)
+        else:
+            pkl_acq_date = None
+            warnings.warn(
+                "Could not parse the acquisition datetime "
+                f"({pkl_raw_acq_date}) found in the following stimulus *.pkl: "
+                f"{self._extractor.get_behavior_stimulus_file()}"
+            )
+
+        if pkl_acq_date:
+            acq_start_diff = (
+                    extractor_acq_date - pkl_acq_date).total_seconds()
+            # If acquisition dates differ by more than an hour
+            if abs(acq_start_diff) > 360:
+                session_id = self._extractor.get_behavior_session_id()
+                warnings.warn(
+                    "The `date_of_acquisition` field in LIMS "
+                    f"({extractor_acq_date}) for behavior session "
+                    f"({session_id}) deviates by more "
+                    f"than an hour from the `start_time` ({pkl_acq_date}) "
+                    "specified in the associated stimulus *.pkl file: "
+                    f"{self._extractor.get_behavior_stimulus_file()}"
+                )
+        return extractor_acq_date
+
+    @property
+    def reporter_line(self) -> Optional[str]:
+        """There can be multiple reporter lines, so it is returned from LIMS
+        as a list. But there shouldn't be more than 1 for behavior. This
+        tries to convert to str
+
+        Returns
+        ---------
+        single reporter line, or None if not possible
+        """
+        reporter_line = self._extractor.get_reporter_line()
+
+        if isinstance(reporter_line, str):
+            return reporter_line
+
+        if len(reporter_line) == 0:
+            warnings.warn('No reporter line')
+            return
```

Comment:
Should we explicitly return `None` here, so that it is obvious that *something* is going to get written to the NWB file?

Reply:
Actually `return` and `return None` are equivalent. I usually explicitly return None; not sure why I did it this way. I will explicitly return None.